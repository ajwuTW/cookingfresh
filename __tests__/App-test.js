import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../App';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
const middlewares = []; // you can mock any middlewares here if necessary
const mockStore = configureStore(middlewares);

// it('renders the loading screen', async () => {
//   const tree = renderer.create(<App />).toJSON();
//   expect(tree).toMatchSnapshot();
// });

const initialState = {
};


describe('Testing App', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <App loadJestSetting />,
      { context: { store: mockStore(initialState) } , adapter: new Adapter()},
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
