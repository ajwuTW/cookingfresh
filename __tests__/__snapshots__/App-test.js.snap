// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppLoading
    onError={[Function]}
    onFinish={[Function]}
    startAsync={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;

exports[`renders the loading screen 1`] = `<ExponentAppLoading />`;
